src/Tickets/Commerce/Hooks.php

```diff
@@ -143,6 +145,54 @@ protected function add_filters() {
 		add_action( 'tribe_editor_config', [ $this, 'filter_tickets_editor_config' ] );
 
 		add_filter( 'wp_list_table_class_name', [ $this, 'filter_wp_list_table_class_name' ], 10, 2 );
+
+		add_filter( 'tribe_dropdown_tec_tc_order_table_events', [ $this, 'provide_events_results_to_ajax' ], 10, 2 );
+	}
+
+	/**
+	 * Provides the results for the events dropdown in the Orders table.
+	 *
+	 * @since TBD
+	 *
+	 * @param array $results The results.
+	 * @param array $search The search.
+	 *
+	 * @return array
+	 */
+	public function provide_events_results_to_ajax( $results, $search ) {
+		if ( empty( $search['term'] ) ) {
+			return $results;
+		}
+
+		$term = $search['term'];
+
+		$args = [
+			'no_found_rows'          => true,
+			'update_post_meta_cache' => false,
+			'update_post_term_cache' => false,
+			'post_type'              => (array) tribe_get_option( 'ticket-enabled-post-types', [] ),
+			'post_status'            => 'any',
+			'posts_per_page'         => 10,
+			's'                      => $term,
+		];
+
+		$query = new WP_Query( $args );
+
+		if ( empty( $query->posts ) ) {
+			return $results;
+		}
+
+		$results = array_map(
+			function ( $r ) {
+				return [
+					'id'   => $r->ID,
+					'text' => apply_filters( 'the_title', $r->post_title ),
```

Comment:
Why not use `get_the_title()`? It handles all the things we need.

Reply:
I am not a 100% if get_the_title would utilize already queried and retrieved object. To avoid that (and a potential double query) i applied the filters in the existing value.

Reply:
If you pass the post ID to `get_the_title()`, then you don't have to worry about it using the correct object.

> To avoid that (and a potential double query)

Within a single request, all queries to post objects are cached. You can see this in `\WP_Post::get_instance()`.

Reply:
Thanks @JPry ! I definitely can pass the post id. In this case i had some doubts since im specifally telling wp to avoid updating meta and term cache since i wasn't going to use them. I guess it has no reason not to cache what has been retrieved so it shouldn't perform a double query for what is already retrieved